admin/services/data-access/pupil-access-arrangements.data.service.js

```diff
@@ -147,19 +147,93 @@ pupilAccessArrangementsDataService.sqFindPupilsWithAccessArrangements = async (d
     }
   ]
   const sql =
-    `SELECT p.urlSlug, p.foreName, p.middleNames, p.lastName, p.dateOfBirth, aa.description
+    `SELECT p.urlSlug, p.foreName, p.middleNames, p.lastName, p.dateOfBirth, aa.description,
+    CASE WHEN pa.id IS NULL THEN 0 ELSE 1 END notTakingCheck,
+    (
+      SELECT CASE WHEN latestCompletedCheckDate IS NULL THEN 0 ELSE 1 END FROM (
+        SELECT ( 
+          SELECT TOP 1 chk.createdAt
+          FROM ${sqlService.adminSchema}.[check] chk
+          WHERE chk.pupil_id = p.id
+          AND chk.checkStatus_id = 3
```

Comment:
I don't think we should rely on knowledge of the database id for state.  Please link in the check status table and use the status code instead.

Reply:
seconded